File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> paren;
        for (char i : s) {
            if (i == '(' || i == '{' || i == '[') {
                paren.push(i);    
            } else if (not paren.empty() && (i == ')' && paren.top() == '(' || i == '}' && paren.top() == '{' || i == ']' && paren.top() == '[')) {
                paren.pop();  
            } else {
                return false;
            }
        }
        if (paren.empty()) {
            return true;
        }
        return false;
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = ""
        for char in s:
            if char in ["(", "{", "["]:
                stack = stack + char
                continue
            if char == ")":
                if len(stack) == 0 or stack[-1] != "(":
                    return False
                stack = stack[:-1]
            if char == "}":
                if len(stack) == 0 or stack[-1] != "{":
                    return False
                stack = stack[:-1]
            if char == "]":
                if len(stack) == 0 or stack[-1] != "[":
                    return False
                stack = stack[:-1]
        return len(stack) == 0